sdk/test-utils/recorder/test/node/recorder.spec.ts

```diff
@@ -0,0 +1,278 @@
+    const fs = require("fs");
+    fs.writeFileSync(
+      "./recordings/node/the_recorders_public_api_on_nodejs/recording_softrecord_should_rerecord_a_simple_outdated_test.js",
+      expectedRecording
+    );
+
+    // We create a very simple HTTP server that serves some content at a specific port.
+    const http = require("http");
+    const server = http.createServer(function(_: any, res: any) {
+      res.write(expectedHttpResponse);
+      res.end();
+    });
+    server.listen(8080);
+
+    const changedTestFunction = () => {
+      let the_contents_have_changed = true;
+      return the_contents_have_changed;
+    };
+
+    // The recorder should start in the beforeEach call.
+    // To emulate that behavior while keeping the test code as contained as possible,
+    // we're compensating with this.
+    const fakeThis: TestContextInterface = new TestContext(this.test! as TestContextTest, {
+      ...this.currentTest,
+      file: __filename,
+      fn: changedTestFunction
+    });
+
+    const recorder = record(fakeThis, recorderEnvSetup);
+
+    const response = await helloWorldRequest();
+
+    // This test's request reached the server and received the expected response.
+    expect(response).to.equal(expectedHttpResponse);
+
+    // Cleaning everything before we continue verifying the results.
+    server.close();
+    recorder.stop();
+
+    // The recorder takes some time to finish writing the output file.
+    // It's not a second, but we're being pessimists.
+    await delay(1000);
+    const recording = fs.readFileSync(
+      "./recordings/node/the_recorders_public_api_on_nodejs/recording_softrecord_should_rerecord_a_simple_outdated_test.js",
+      { encoding: "utf-8" }
+    );
+
+    // Nock does store the date of the request. Let's strip that from the response.
+    const recordingWithoutDate = recording.replace(/Date',\n[^\n]*\n/, "Date',\n  'DATE',\n");
+
+    // Let's make a new expected recording variable,
+    // this time with the new hash.
+    const expectedRecordingWithUpdatedHash = recordingWithoutDate.replace(
+      expectedHash,
+      MD5(changedTestFunction.toString())
+    );
+
+    // The hash has changed in the recorded file.
+    expect(recordingWithoutDate).to.equal(expectedRecordingWithUpdatedHash);
+  });
+
+  it("soft-record should skip a simple unchanged test", async function() {
+    process.env.TEST_MODE = "soft-record";
+
+    // Making sure the expected recording actually exists before running playback.
+    const fs = require("fs");
+    fs.writeFileSync(
+      "./recordings/node/the_recorders_public_api_on_nodejs/recording_softrecord_should_skip_a_simple_unchanged_test.js",
+      expectedRecording
+    );
+
+    // The recorder should start in the beforeEach call.
+    // To emulate that behavior while keeping the test code as contained as possible,
+    // we're compensating with this.
+    (this as any).currentTest = {
```

Comment:
again, would be good to not munge the state of the test we're excuting if at all possible

Reply:
my bad again 😁 I thought I fixed this too.